pkg/retrieval/retrieval.go

```diff
@@ -364,7 +364,7 @@ func (s *Service) prepareCredit(ctx context.Context, peer, chunk swarm.Address,
 // retrieve request.
 func (s *Service) closestPeer(addr swarm.Address, skipPeers []swarm.Address, allowUpstream bool) (swarm.Address, error) {
 
-	closest, err := s.peerSuggester.ClosestPeer(addr, false, topology.Select{Reachable: true}, skipPeers...)
+	closest, err := s.peerSuggester.ClosestPeer(addr, false, topology.Select{Reachable: true, Healthy: true}, skipPeers...)
```

Comment:
Is this required? Are we seeing failed retrieval attempts?

I am just afraid of restricting to only healthy peers here. We could have reachable peers who could potentially respond but wont be considered.

Reply:
On occasion, yes, there are failed downloads, but still a very small rate, like two or three failed downloads per day, out of like 500 smoke test downloads. the other issue is download speed is lagging behind uploads, which is counterintuitive. 

adding healthy peer checks to pushsync improved retrieval as well but adding this check to retrieval is the next step.

using only healthy peers would boost performance and reliability.

Reply:
we can put this on hold until after the localstore release if you'd like